tests/components/probe_plus/test_config_flow.py

```diff
@@ -0,0 +1,141 @@
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.service_info.bluetooth import BluetoothServiceInfo
+
+from tests.common import MockConfigEntry
+
+service_info = BluetoothServiceInfo(
+    name="FM210 aa:bb:cc:dd:ee:ff",
+    address="aa:bb:cc:dd:ee:ff",
+    rssi=-63,
+    manufacturer_data={},
+    service_data={},
+    service_uuids=[],
+    source="local",
+)
+
+
+@pytest.fixture
+def mock_discovered_service_info() -> Generator[AsyncMock]:
+    """Override getting Bluetooth service info."""
+    with patch(
+        "homeassistant.components.probe_plus.config_flow.async_discovered_service_info",
+        return_value=[service_info],
+    ) as mock_discovered_service_info:
+        yield mock_discovered_service_info
+
+
+async def test_user_config_flow_creates_entry(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_discovered_service_info: AsyncMock,
+) -> None:
+    """Test the user configuration flow successfully creates a config entry."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ADDRESS: "aa:bb:cc:dd:ee:ff",
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        CONF_ADDRESS: "aa:bb:cc:dd:ee:ff"
+    }
+
+async def test_user_flow_already_configured(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_setup_entry: AsyncMock,
+    mock_discovered_service_info: AsyncMock,
+) -> None:
+    """Test that the user flow aborts when the entry is already configured."""
+    mock_config_entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ADDRESS: "aa:bb:cc:dd:ee:ff",
+        },
+    )
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+
+async def test_bluetooth_discovery(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_discovered_service_info: AsyncMock,
+) -> None:
+    """Test we can discover a device."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_BLUETOOTH}, data=service_info
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "bluetooth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_ADDRESS: "aa:bb:cc:dd:ee:ff",
+        },
```

Comment:
Also, we discover the device via bluetooth, why do we still have to enter the address?

Reply:
Was an error on my part when defining the tests. You are correct - this is not needed.